### PR TITLE
Feature/security

### PR DESCRIPTION
**Pull request by julianvasquez-wolox**
## Summary
- Security dependencies added to gradle
- Security Config and Custom Autenthication Provider for basic authentication added
- Integration tests 
- Unit tests for Authentication Provider

## Trello Card
- https://trello.com/c/3DJDwSpB/23-integrar-spring-security

### CODE REVIEW

File: src/main/java/wolox/training/TrainingApplication.java [hunk 1]
```diff
@@ -2,12 +2,20 @@
...
+import org.springframework.context.annotation.Bean;
 import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
 
 @SpringBootApplication
 @EnableJpaRepositories
 public class TrainingApplication {
 
+	@Bean
```
Comment (facucompe) [inline src/main/java/wolox/training/TrainingApplication.java]: Let's move this Bean to a @Configuration class

File: src/test/java/wolox/training/security/CustomAuthenticationProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  @Autowired
+  CustomAuthenticationProvider customAuthenticationProvider;
+
+  @Test
+  public void whenAuthenticateWithValidUserMatchingPassword_thenReturnToken(){
+    user = userFactory.createTestUser();
+    authentication = new UsernamePasswordAuthenticationToken(user.getUsername(), user.getPassword(), new ArrayList<>());
+    given(userRepository.findByUsername(authentication.getName())).willReturn(Optional.ofNullable(user));
+    given(passwordEncoder.matches(authentication.getCredentials().toString(),user.getPassword())).willReturn(true);
```
Comment (facucompe) [inline src/test/java/wolox/training/security/CustomAuthenticationProviderTest.java:48]: verify this mock is being called

File: src/test/java/wolox/training/security/CustomAuthenticationProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    assertTrue(authResponse.isAuthenticated());
+  }
+
+  @Test
+  public void whenAuthenticateWithValidUserNotMatchingPassword_thenReturnToken(){
+    user = userFactory.createTestUser();
+    authentication = new UsernamePasswordAuthenticationToken(user.getUsername(), user.getPassword(), new ArrayList<>());
+    given(userRepository.findByUsername(authentication.getName())).willReturn(Optional.ofNullable(user));
+    given(passwordEncoder.matches(authentication.getCredentials().toString(),user.getPassword())).willReturn(false);
```
Comment (facucompe) [inline src/test/java/wolox/training/security/CustomAuthenticationProviderTest.java:62]: Same here

**Review by facucompe — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/TrainingApplication.java (hunk 1 above)
Comment (facucompe) [reply src/main/java/wolox/training/TrainingApplication.java]: hahah, i tag a github user

**Review by facucompe — APPROVED** [review]

**Review by rabinovichm — APPROVED** [review]

**Pull request merged**